tests/models/test_layout.py

```diff
@@ -19,25 +19,198 @@
 
 import os
 import unittest
-import datetime
+import tempfile
+import shutil
+
 from in_toto.models.layout import Layout, Step, Inspection
 from in_toto.models.metadata import Metablock
 import in_toto.models.link
 import in_toto.exceptions
 import in_toto.verifylib
 import securesystemslib.exceptions
 
+
+class TestLayoutMethods(unittest.TestCase):
+  """Test Layout methods. """
+
+  @classmethod
+  def setUpClass(self):
+    """Create temporary test directory and copy gpg keychain, and rsa keys from
+    demo files. """
+    self.gpg_keyid1 = "7b3abb26b97b655ab9296bd15b0bd02e1c768c43"
+    self.gpg_keyid2 = "8288ef560ed3795f9df2c0db56193089b285da58"
+    self.test_dir = os.path.realpath(tempfile.mkdtemp())
+
+    # Create directory to run the tests without having everything blow up
+    self.working_dir = os.getcwd()
+
+    # Copy GPG keyring to temp test dir
+    tests_dir = os.path.join(os.path.dirname(os.path.realpath(__file__)), "..")
+
+    self.gnupg_home = os.path.join(self.test_dir, "rsa")
+    shutil.copytree(
+        os.path.join(tests_dir, "gpg_keyrings", "rsa"),
+        self.gnupg_home)
+
+    # Copy keys to temp test dir
+    key_names = ["bob", "bob.pub", "carl.pub"]
+    for name in key_names:
+      shutil.copy(
+        os.path.join(tests_dir, "demo_files", name),
+        os.path.join(self.test_dir, name))
+
+    self.key_path = os.path.join(self.test_dir, "bob")
+    self.pubkey_path1 = os.path.join(self.test_dir, "bob.pub")
+    self.pubkey_path2 = os.path.join(self.test_dir, "carl.pub")
+
+    os.chdir(self.test_dir)
+
+
+
+  @classmethod
+  def tearDownClass(self):
+    """Change back to initial working dir and remove temp test directory. """
+    os.chdir(self.working_dir)
+    shutil.rmtree(self.test_dir)
+
+
+  def test_set_relative_expiration(self):
+    """Test adding expiration date relative from today. """
+    layout = Layout()
+    with self.assertRaises(securesystemslib.exceptions.FormatError):
+      layout.set_relative_expiration(days=None, months=0, years=0)
+
+    with self.assertRaises(securesystemslib.exceptions.FormatError):
+      layout.set_relative_expiration(days=0, months="", years=0)
+
+    with self.assertRaises(securesystemslib.exceptions.FormatError):
+      layout.set_relative_expiration(days=0, months=0, years=[])
+
+    layout.set_relative_expiration(days=1)
+    layout._validate_expires()
+    layout.set_relative_expiration(months=2)
+    layout._validate_expires()
+    layout.set_relative_expiration(years=3)
+    layout._validate_expires()
+    layout.set_relative_expiration(days=3, months=2, years=1)
+    layout._validate_expires()
```

Comment:
can you throw some negative numbers in here to see what happens? :)

Reply:
Sure, will do.